docs/concepts.md

```diff
@@ -0,0 +1,153 @@
+# Concepts
+
+PrairieLearn is extremely flexible and powerful, but this flexibility can appear opaque at first. This document provides an overview of the key concepts in PrairieLearn, and how they fit together.
+
+## High-level overview
+
+A [**course**](./course.md) in PrairieLearn is composed of **course instances** and **questions**. Each [course instance](./courseInstance.md) is an offering of a course in a particular semester/term. For example, a course "CS 225" might have instances "CS 225 Spring 2022" and "CS 225 Fall 2022". A course instance contains all the **assessments** for that particular term. The course itself contains the questions that are shared across all instances. You can think of the course as containing a bank of questions that are then used to create assessments for each instance of the course, and a way to organize every instance of the course.
+
+=== "Concept Map"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map.d2"
```

Comment:
Honestly I'd be fine with making a directory per docs page by default. This would make all pages the same and allow us to move all images Into the per-page directories.

Reply:
I'm a little hesitant right now, as it makes hyperlinking a little harder. Not sure I see a major payoff until pages have more images. We only have 5 images across all non-concept pages currently.

Reply:
I'm fine only doing it when we have images.

Reply:
Done.